src/coffee-script.coffee

```diff
@@ -157,7 +157,8 @@ exports.eval = (code, options = {}) ->
       sandbox.module  = _module  = new Module(options.modulename || 'eval')
       sandbox.require = _require = (path) ->  Module._load path, _module, true
       _module.filename = sandbox.__filename
-      _require[r] = require[r] for r in Object.getOwnPropertyNames require when r isnt 'paths'
+      for r in Object.getOwnPropertyNames require when [ 'paths', 'arguments', 'caller' ].indexOf(r) is -1
```

Comment:
nice, I thought there was a better way but couldn't find it in a quick scan of the docs, thanks